trinity/plugins/eth2/network_generator/plugin.py

```diff
@@ -0,0 +1,177 @@
+        self.validator_keys_dir = self.client_dir / VALIDATOR_KEY_DIR
+
+
+class NetworkGeneratorPlugin(BaseMainProcessPlugin):
+    @property
+    def name(self) -> str:
+        return "NetworkGenerator"
+
+    def configure_parser(self, arg_parser: ArgumentParser, subparser: _SubParsersAction) -> None:
+
+        testnet_generator_parser = subparser.add_parser(
+            'testnet',
+            help='Generate testnet files',
+        )
+        testnet_generator_parser.add_argument(
+            "--network-dir",
+            help="Directory to create all the files into",
+            type=Path,
+            default=Path("."),
+        )
+        testnet_generator_parser.add_argument(
+            "--num",
+            help="Number of validators to generate",
+            type=int,
+            default=100,
+        )
+        testnet_generator_parser.add_argument(
+            "--genesis-delay",
+            help="Seconds before genesis time from now",
+            type=int,
+            default=60,
+        )
+
+        testnet_generator_parser.set_defaults(func=self.run_generate_testnet_dir)
+
+    def run_generate_testnet_dir(self, args: Namespace, trinity_config: TrinityConfig) -> None:
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(self._run_generate_testnet_dir(args))
+        loop.close()
+
+    async def _run_generate_testnet_dir(self, args: Namespace)-> None:
+        self.logger.info("Generating testnet")
+        self.network_dir = args.network_dir
+        if len(os.listdir(self.network_dir)) > 0:
+            self.logger.error("This directory is not empty, won't create network files here.")
+            sys.exit(1)
+
+        self.generate_trinity_root_dirs()
+        self.generate_keys(args.num)
+        self.generate_genesis_state(args.genesis_delay)
+
+        self.logger.info(bold_green("Network generation completed"))
+
+    def generate_keys(self, num: int) -> None:
+        self.logger.info(f"Creating {num} validators' keys")
+        self.keys_dir = self.network_dir / KEYS_DIR
+        self.keys_dir.mkdir()
+
+        privkeys = tuple(int.from_bytes(
+            hash_eth2(str(i).encode('utf-8'))[:4], 'big')
+            for i in range(num)
+        )
+        self.keymap = {bls.privtopub(key): key for key in privkeys}
+
+        num_of_clients = len(self.clients)
+        for validator_index, key in enumerate(privkeys):
+            file_name = f"v{validator_index:07d}.privkey"
+            private_key_path = self.keys_dir / file_name
+            with open(private_key_path, "w") as f:
+                f.write(str(key))
+
+            # Distribute keys to clients
+            client = self.clients[validator_index % num_of_clients]
+            with open(client.validator_keys_dir / file_name, "w") as f:
+                f.write(str(key))
+
+    def generate_genesis_state(self, genesis_delay: Second) -> None:
+        state_machine = XiaoLongBaoStateMachine
```

Comment:
Would it be more explicit that this is a class?
```suggestion
        state_machine_class = XiaoLongBaoStateMachine
```